packages/features/bookings/Booker/components/OverlayCalendar/OverlayCalendarContainer.tsx

```diff
@@ -110,9 +109,16 @@ function OverlayCalendarSwitch({
 export function OverlayCalendarContainer() {
   const isEmbed = useIsEmbed();
   const searchParams = useSearchParams();
-  const [continueWithProvider, setContinueWithProvider] = useState(false);
-  const [calendarSettingsOverlay, setCalendarSettingsOverlay] = useState(false);
-  const { data: session } = useSession();
+  const [continueWithProvider, setContinueWithProvider] = useOverlayCalendarStore(
+    (state) => [state.continueWithProviderModal, state.setContinueWithProviderModal],
+    shallow
+  );
+  const [calendarSettingsOverlay, setCalendarSettingsOverlay] = useOverlayCalendarStore(
+    (state) => [state.calendarSettingsOverlayModal, state.setCalendarSettingsOverlayModal],
+    shallow
+  );
```

Comment:
Using shallow state from zustand so the api call doesnt block threads 

Probably better ways of solving this but this is good for a hot fix